app/uk/gov/hmrc/agentsubscriptionfrontend/service/SubscriptionService.scala

```diff
@@ -61,4 +82,50 @@ class SubscriptionService @Inject()(agentSubscriptionConnector: AgentSubscriptio
       case e => throw e
     }
   }
+
+  def completePartialSubscription(details: CompletePartialSubscriptionBody)(
+    implicit hc: HeaderCarrier,
+    ec: ExecutionContext): Future[(Boolean, Option[Arn])] =
+    for {
+      subscriptionProgress <- getSubscriptionStatus(details.utr, details.knownFacts.postcode) //FixMe APB-2805 this check should happen previously and can be stored in the session
+      isPartiallySubscribed = subscriptionProgress.state == SubscriptionState.IsOnlySubscribedInETMP
+      (wasPartiallySubscribed, optSucceededAllocatingArn) <- isPartiallySubscribed match {
+                                                              case true =>
+                                                                agentSubscriptionConnector
+                                                                  .completePartialSubscription(details)
+                                                                  .map(arn => (true, Some.apply(arn)))
+                                                                  .recover {
+                                                                    case e: Upstream4xxResponse
+                                                                        if Seq(Status.FORBIDDEN, Status.CONFLICT) contains e.upstreamResponseCode =>
+                                                                      (true, None)
+                                                                    case e => throw e
+                                                                  }
+                                                              case false =>
+                                                                Future successful (false, None)
+                                                            }
+    } yield (wasPartiallySubscribed, optSucceededAllocatingArn)
+
+  def getSubscriptionStatus(utr: Utr, postcode: String)(
+    implicit hc: HeaderCarrier,
+    ec: ExecutionContext): Future[SubscriptionProcess] =
+    agentSubscriptionConnector.getRegistration(utr, postcode).map {
+
+      case Some(reg) if reg.isSubscribedToAgentServices && reg.isSubscribedToETMP =>
```

Comment:
You can remove reg.isSubscribedToETMP from here since isSubscribedToAgentServices contains both.

Reply:
ok, kept wonering whether ETMP record means anything if theyre already subscribed.